# 가설검정 연습문제 (2)

In [1]:
from scipy import stats
from pandas import read_excel

## 문제 1

어떤 화학약품의 제조에 상표가 다른 두 종류의 원료가 사용되고 있다.

각 원료에서 주성분 A의 함량은 다음과 같다. 단, 함량은 정규분포를 따른다고 가정한다.

이 두 원료의 주성분 A의 함량이 다른지를 분석하라.

![img](res/q01.png)

> https://data.hossam.kr/E02/material.xlsx

In [2]:
df1 = read_excel('https://data.hossam.kr/E02/material.xlsx')
df1

상표1   상표2
0   80.4  80.1
1   78.2  81.2
2   80.1  79.5
3   77.1  78.0
4   79.6  76.1
5   80.4  77.0
6   81.6  80.1
7   79.9  79.9
8   84.4  78.8
9   80.9  80.8
10  83.1   NaN

In [3]:
var1 = df1['상표1'].var()
var2 = df1['상표2'].var()
var1,var2

(4.153636363636368, 2.7538888888888917)

In [5]:
df1_1 = df1.drop(index=10)
df1_1

상표1   상표2
0  80.4  80.1
1  78.2  81.2
2  80.1  79.5
3  77.1  78.0
4  79.6  76.1
5  80.4  77.0
6  81.6  80.1
7  79.9  79.9
8  84.4  78.8
9  80.9  80.8

귀무가설 : 두 원료의 주성분 A의 함량에는 차이가 없다.

대립가설 : 두 원료의 주성분 A의 함량에는 차이가 있다.

In [6]:
stats.ttest_ind(df1_1['상표1'],df1_1['상표2'])

TtestResult(statistic=1.3710671216714043, pvalue=0.18720759661580608, df=18.0)

결과 해석

차이값에 대한 t 통계량의 값이 1.37, 대응되는 유의확률이 0.18로 0.05보다 적으므로 유의수준 0.05에서 귀무가설을 기각, 주 성분 A의 함량에는 차이가 없지 않다고 결론 내릴 수 있다. 

## 문제 2

특정 피임약이 사용자의 혈압을 저하시키는지 조사하고자 한다.

이를 위해 부인 15명을 대상으로 평상시 혈압을 측정한 뒤, 이들에게 이 피임약을 일정 기간 복용하게 한 후 이들의 혈압을 다시 측정한 결과를 기록했다.

얻어진 데이터는 다음과 같다. 피임약 복용이 혈압에 영향을 주는지 분석하라.

![img](res/q02.png)

> https://data.hossam.kr/E02/blood_pressure.xlsx


In [7]:
df2 = read_excel('https://data.hossam.kr/E02/blood_pressure.xlsx')
df2

복용전  복용후
0    70   68
1    80   72
2    72   62
3    76   70
4    76   58
5    76   66
6    72   68
7    78   52
8    82   64
9    64   72
10   74   74
11   92   60
12   74   74
13   68   72
14   84   74

귀무가설 : 피임약 복용이 혈압에 영향을 미치지 않는다.

대립가설 : 피임약 복용이 혈압에 영향을 미친다.

In [8]:
stats.ttest_rel(df2['복용전'],df2['복용후'],alternative='two-sided')

TtestResult(statistic=3.105360487466109, pvalue=0.007749436106654119, df=14)

결과 해석

차이값에 대한 t 통계량의 값이 3.10, 대응되는 유의확률이 0.0077로 0.05보다 적으므로 유의수준 0.05에서 귀무가설을 기각, 대립가설을 채택해 피임약 복용이 혈압에 영향을 미친다고 결론을 내릴 수 있다. 

## 문제3

고등학교 육상선수에게 체중감량을 시키면 달리기에 어떤 변화가 일어나는지 조사하기 위
하여, 어떤 고등학교 육상선수 10명에게 감량훈련(2~4kg 정도)을 시킨 후 100m 달리기를
실시하여 다음의 데이터를 얻었다.

감량 전과 후에 달리기 속도에 차이가 있는지 분석하라

![img](res/q03.png)

> https://data.hossam.kr/E02/runner_diet.xlsx


In [10]:
df3 = read_excel('https://data.hossam.kr/E02/runner_diet.xlsx')
df3

감량전   감량후    
0  14.5  14.3 NaN
1  13.4  13.4 NaN
2  13.0  13.1 NaN
3  14.5  13.9 NaN
4  13.3  13.1 NaN
5  14.2  13.9 NaN
6  13.6  13.6 NaN
7  13.9  14.0 NaN
8  13.2  12.8 NaN
9  13.0  12.8 NaN

귀무가설 : 감량 전과 후에 달리기 속도에 차이가 없다. 

대립가설 : 감량 전과 후에 달리기 속도에 차이가 있다. 

In [11]:
stats.ttest_rel(df3['감량전'],df3['감량후'],alternative='two-sided')

TtestResult(statistic=2.375306807446375, pvalue=0.04154604923393679, df=9)

결과 해석

차이값에 대한 t 통계량의 값이 2.375, 대응되는 유의확률이 0.04로 0.05보다 적으므로 유의수준 0.05에서 귀무가설을 기각, 대립가설을 채택해 감령 전과 후 달리기 속도에 차이가 있다고 말할 수 있다. 

## 문제 4

고등학교 레슬링 선수에게 체중감량을 시키면 체력에 어떤 변화가 일어나는지 조사하기 위하여, 어느 고등학교의 레슬링 선수 12명을 표본추출하여 감량시킨 후 다음의 체력 데이터를 얻었다.

감량 전과 감량 후 간에 차이가 있는가의 대응비교를 악력(우), 윗몸일으키기, 턱걸이에 대하여 실시하고 분석하라.

![img](res/q04.png)

> https://data.hossam.kr/E02/wrestler_diet.xlsx

In [79]:
df4 = read_excel('https://data.hossam.kr/E02/wrestler_diet.xlsx')
df4

이름  감량전 악력  감량후 악력  감량전 윗몸일으키기  감량후 윗몸일으키기  감량전 턱걸이  감량후 턱걸이
0   B.S      43      41          35          41       25       29
1   P.J      42      41          40          44       25       29
2   G.G      52      50          36          41       22       23
3   A.L      53      52          38          42       10       15
4   I.H      44      40          36          42       15       19
5   B.Y      46      47          39          44       17       19
6   I.K      48      47          35          42       24       26
7   K.B      58      55          34          36       17       18
8   K.J      50      50          33          35       17       20
9   S.S      48      50          39          45       21       25
10  S.P      56      48          37          37       31       35
11  S.J      43      41          33          37       15       22

In [83]:
for i in range(1,len(df4.columns),2) :
    statistic,pvalue = stats.ttest_rel(df4[df4.columns[i]],df4[df4.columns[i+1]],alternative='two-sided')
    # print(statistic,pvalue)
    print('귀무가설 : {columns1}와(과) {columns2}에 차이가 없다.'.format(columns1=df4.columns[i],columns2=df4.columns[i+1]))
    print('대립가설 : {columns1}와(과) {columns2}에 차이가 있다.'.format(columns1=df4.columns[i],columns2=df4.columns[i+1]))

    if pvalue < 0.05 :
        print('결과 해석 : 차이값에 대한 t 통계량의 값이 {statistic}, 대응되는 유의확률이 {pvalue}로 0.05보다 적으므로 귀무가설을 기각, 대립가설을 채택해 {columns1}와(과) {columns2}에 차이가 있다고 말할 수 있다.'.format(statistic=round(statistic,3),pvalue=round(pvalue,3),columns1=df4.columns[i],columns2=df4.columns[i+1]))
    else :
        print('결과 해석 : 차이값에 대한 t 통계량의 값이 {statistic}, 대응되는 유의확률이 {pvalue}로 0.05보다 크므로 통계적으로 유의미한 차이를 발견할 수 없었다.'.format(statistic=round(statistic,3),pvalue=round(pvalue,5),columns1=df4.columns[i],columns2=df4.columns[i+1]))


귀무가설 : 감량전 악력와(과) 감량후 악력에 차이가 없다.
대립가설 : 감량전 악력와(과) 감량후 악력에 차이가 있다.
결과 해석 : 차이값에 대한 t 통계량의 값이 2.365, 대응되는 유의확률이 0.037로 0.05보다 적으므로 귀무가설을 기각, 대립가설을 채택해 감량전 악력와(과) 감량후 악력에 차이가 있다고 말할 수 있다.
귀무가설 : 감량전 윗몸일으키기와(과) 감량후 윗몸일으키기에 차이가 없다.
대립가설 : 감량전 윗몸일으키기와(과) 감량후 윗몸일으키기에 차이가 있다.
결과 해석 : 차이값에 대한 t 통계량의 값이 -7.18, 대응되는 유의확률이 0.0로 0.05보다 적으므로 귀무가설을 기각, 대립가설을 채택해 감량전 윗몸일으키기와(과) 감량후 윗몸일으키기에 차이가 있다고 말할 수 있다.
귀무가설 : 감량전 턱걸이와(과) 감량후 턱걸이에 차이가 없다.
대립가설 : 감량전 턱걸이와(과) 감량후 턱걸이에 차이가 있다.
결과 해석 : 차이값에 대한 t 통계량의 값이 -6.842, 대응되는 유의확률이 0.0로 0.05보다 적으므로 귀무가설을 기각, 대립가설을 채택해 감량전 턱걸이와(과) 감량후 턱걸이에 차이가 있다고 말할 수 있다.
